time timestamp
float32 angle
